docs/source/execute_api.rst

```diff
@@ -0,0 +1,145 @@
+Executing notebooks
+===================
+
+In this section we show how to execute a ``.ipynb`` notebook
+document saving the result in notebook format.
+To export the notebook to other formats see the
+:ref:`next section <nbconvert_library>`.
+
+Executing notebooks programmatically is useful, for example, as a test layer
+in python libraries that include example notebooks, or as a way to
+automate the data analysis in projects involving more than one notebook.
+
+The same functionality of executing notebooks is exposed through a
+:ref:`command line interface <commandline_usage>` or a python API interface.
+In this section we will (mostly) illustrate how to use the python API interface.
```

Comment:
Since the title of the page is just 'executing notebooks', I think it should at least briefly mention the `--execute` command line flag.